inc/Service/BannersService.php

```
@@ -0,0 +1,97 @@
+<?php
+
+namespace Vichan\Service;
+
+use Vichan\Data\Driver\CacheDriver;
+
+class BannersService {
+	private const BANNERS_DIR = 'static/banners/%s/';
+	private const PRIORITY_DIR = 'static/banners_priority/';
+	private const ALLOWED_EXT = ['jpg', 'jpeg', 'png', 'gif', 'webp'];
+	private const UKKO = 'ukko';
+	private CacheDriver $cache;
+
+	public function __construct(CacheDriver $cache) {
+		$this->cache = $cache;
+	}
+
+	private function getFilesInDirectory(string $dir): array {
+		if (!\is_dir($dir)) {
```

This silent fail going to bite someone. I suggest adding an error log